server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java

```diff
@@ -142,17 +140,16 @@ public void onFailure(String source, Exception e) {
             ActionRequestBuilder<Req, Res> req) throws Exception {
         // Wait for no publication in progress to not accidentally cancel a publication different from the one triggered by the given
         // request.
-        assertBusy(
-            () -> {
-                assertFalse(((Coordinator) internalCluster().getCurrentMasterNodeInstance(Discovery.class)).publicationInProgress());
-                assertThat(StreamSupport.stream(
-                    internalCluster().getInstances(Discovery.class).spliterator(), false)
-                    .map(coordinator -> ((Coordinator) coordinator).getLastAcceptedState().version())
-                    .distinct().toArray(), arrayWithSize(1));
-            });
+        final Coordinator masterCoordinator = (Coordinator) internalCluster().getCurrentMasterNodeInstance(Discovery.class);
+        assertBusy(() -> {
+            assertFalse(masterCoordinator.publicationInProgress());
```

Comment:
Made this whole thing a little less weird to read as well here, the stream + iterator + array size assertion was just weirdly complicated.